### PR TITLE
Predictor for Model

### PR DESCRIPTION
**Pull request by sarthakvarora**

### CODE REVIEW

File: climate_scanner/coarse_article_classifier/coarse_class.py [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+    model = model.to(device)
+    model.eval()
+    tokenizer_output = tokenizer(text, return_tensors="pt")
+    input_ids, attn_mask = tokenizer_output["input_ids"], tokenizer_output["attention_mask"]
+    prob = model(input_ids, attn_mask)
+    pred_label = convert_probs_to_labels(prob, threshold)
+    
+    return pred_label
+
+# Final Predictor
```
Comment (Eliotfrazier) [inline climate_scanner/coarse_article_classifier/coarse_class.py:356]: Package this into a test perhaps?

File: climate_scanner/coarse_article_classifier/predict.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        self.bert_layer = BertModel.from_pretrained(bert_variant)
+        self.output_layer = nn.Linear(d_hidden, 1)
+        self.sigmoid_layer = nn.Sigmoid()
+        
+    def forward(self, input_ids, attn_mask):
+        output = self.bert_layer(input_ids, attention_mask = attn_mask).pooler_output
+        output = self.sigmoid_layer(self.output_layer(output))
+        return output.squeeze(-1)
+
+def predict_text(text, model, tokenizer, threshold = 0.5, device = "cpu"):
```
Comment (Eliotfrazier) [inline climate_scanner/coarse_article_classifier/predict.py:31]: Identical to the function in coarse_class.py?

File: climate_scanner/coarse_article_classifier/predict.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+def load_weights(model, path):
+    path1 = os.path.join(_ROOT, path)
+    model.load_state_dict(torch.load(path1))
+    return model
+
+def convert_probs_to_labels(probs, threshold = 0.5):
+    labels = np.where(probs > threshold, 1, 0)
+    return labels
+
+class BertClassifierModel(nn.Module):
```
Comment (Eliotfrazier) [inline climate_scanner/coarse_article_classifier/predict.py:19]: Again, is this not functionally identifcal to the version in the coarse_class.py, why not just import the class from the other file?

**Review by Eliotfrazier — APPROVED** [review]
Looks good overall, a few ascetic comments, but nothing huge to address...

File: climate_scanner/coarse_article_classifier/coarse_class.py [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+import torch
+import torch.nn as nn
+from torch.optim import Adam
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn.model_selection import train_test_split
+import copy
+import tqdm
+
+df = pd.read_csv('/content/drive/MyDrive/non_climate_data.csv')
```
[the remark this answers is not in the document]
Comment (sarthakvarora) [reply climate_scanner/coarse_article_classifier/coarse_class.py:16]: yes, that's the plan. referenced some other AI4G codes to add this. Also, this is the code safekeeping file, the main file which will be used to predict is the [predict.py](https://github.com/sarthakvarora/climate-scanner/blob/master/climate_scanner/coarse_article_classifier/predict.py) file. Can you review that also? It has elements from this, and is much shorter, so wouldn't take long

File: climate_scanner/coarse_article_classifier/coarse_class.py [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+            true_negatives += torch.sum(torch.isnan(confusion_vector)).item()
+            false_negatives += torch.sum(confusion_vector == 0).item()
+        accuracy = accuracy / len(test_dataloader)
+        recall = true_positives/(true_positives+false_negatives)
+        precision = true_positives/(true_positives+false_positives)
+        f1_score = 2 * precision * recall / (precision + recall)
+    
+    return precision, recall, f1_score, accuracy
+
+def evaluate(model, test_dataloader, threshold = 0.5, device = "cpu"):
```
[the remark this answers is not in the document]
Comment (sarthakvarora) [reply climate_scanner/coarse_article_classifier/coarse_class.py:230]: this is used at one place in train(), thought i'll keep it? I've made the rest of the changes

**Pull request merged**